### DISCUSSION TITLE
Multi-row inserts with queryRaw - getting 0x00 byte

### DISCUSSION THREAD

**Discussion by revmischa** [prisma/prisma#11640] [Q&A] [answered]
DB: Postgres

Code:
```ts
this.prisma.$queryRaw`
      INSERT INTO "CandidateSkill"
      ("candidateId", "skillId", "seniority")
      VALUES ${Prisma.join(
        skills.map((s) => Prisma.sql`(${candidateId}, ${skillNameToIdMap[s.name]}, ${s.experience})`)
      )}
      ON CONFLICT ("candidateId", "skillId") DO UPDATE
      SET "seniority"=EXCLUDED."seniority"`
```

Generated query:
```sql
INSERT INTO "CandidateSkill"
          ("candidateId", "skillId", "seniority")
          VALUES ($1, $2, $3),($4, $5, $6)
          ON CONFLICT ("candidateId", "skillId") DO UPDATE
          SET "seniority"=EXCLUDED."seniority"
```

Error:
`Raw query failed. Code: `22021`. Message: `db error: ERROR: invalid byte sequence for encoding “UTF8”: 0x00`

The result of `Prisma.join`:
```ts
[
      {
        text: '($1, $2, $3)',
        sql: '(?, ?, ?)',
        values: [
          'a5fab966-a225-4095-bf3e-f67fb71ef34f',
          '5df1ae31-142a-44ff-bca7-7b2db71f30de',
          0.5
        ]
      },
      {
        text: '($1, $2, $3)',
        sql: '(?, ?, ?)',
        values: [
          'a5fab966-a225-4095-bf3e-f67fb71ef34f',
          '9cc5c898-b791-4838-9445-107ddd4419c7',
          0.75
        ]
      }
    ]
```

I don't even know how to begin debugging this error.

**Comment by revmischa** [comment]
I tried switching to local MacOS Postgres.app so I could more easily view what's going on and it ... works. 

When I test against the `postgres:11` docker image, it errors with the above error.

Spooky! 👻

**Comment by revmischa** [comment, marked as answer]
If I test against local docker `postgres:11` I get the above error, if I test with v12 or on Aurora Serverless I don't. I guess that's the workaround for now but it's concerning because I believe Aurora Serverless is on pg 11 (compatible).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
